test/test_tensor.h

```diff
@@ -926,14 +926,25 @@ TEST(tensor, nd1) {
 
 }
 
-TEST(tensor, print) {
-    Tensor<float_t, 4> t({3,2,2,2});
+template<size_t T> std::ostream& print_tester(std::ostream &os) {
+    os << "\nPrinting Tensor of size " << T << ":\n\n";
```

Comment:
At a glance, it's not very clear to me what the `size` means in here.

```
    os << "\nPrinting " << T << "-dimensional Tensor" << ":\n\n";
```
I think above message would explain more about the Tensor.